.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  release:
+    types: [created]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        scala_version: &scala_versions
+          - 2.12.12
+          - 2.13.4
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with: &cache_options
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+            ~/.coursier
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ matrix.scala_version }}
+          restore-keys:
+            - ${{ runner.os }}-sbt-${{ hashFles('**/*.sbt') }}-
+      - name: Build
+        run: sbt ++${{ matrix.scala_version }} compile packageBin packageDoc packageSrc
+      - name: Test
+        run: sbt ++${{ matrix.scala_version }} test
+      - uses: actions/upload-artifact@v2
+        with:
+          name: target-${{ matrix.scala_version }}
+          path: '*/target/'
+          retention-days: 1
+  publish:
+    if: github.event_name == "push" || github.event_name == "release"
```

Comment:
similarly, should this be on tag/master push?

Reply:
this workflow is only triggered on pushes to master and when releases are created.

Reply:
Right, should the workflow trigger on release-create or on tag push?